ivy/func_wrapper.py

```diff
@@ -365,30 +365,22 @@ def _wrap_function(key: str, to_wrap: Callable, original: Callable) -> Callable:
                 )
         return to_wrap
     if isinstance(to_wrap, FunctionType):
-        if hasattr(original, "array_spec"):
-            to_wrap.array_spec = original.array_spec
-        if hasattr(original, "infer_device") and not hasattr(to_wrap, "infer_device"):
-            to_wrap = infer_device(to_wrap)
-        if hasattr(original, "infer_dtype") and not hasattr(to_wrap, "infer_dtype"):
-            to_wrap = infer_dtype(to_wrap)
-        if hasattr(original, "outputs_to_ivy_arrays") and not hasattr(
-            to_wrap, "outputs_to_ivy_arrays"
-        ):
-            to_wrap = outputs_to_ivy_arrays(to_wrap)
-        if hasattr(original, "inputs_to_native_arrays") and not hasattr(
-            to_wrap, "inputs_to_native_arrays"
-        ):
-            to_wrap = inputs_to_native_arrays(to_wrap)
-        if hasattr(original, "inputs_to_ivy_arrays") and not hasattr(
-            to_wrap, "inputs_to_ivy_arrays"
-        ):
-            to_wrap = inputs_to_ivy_arrays(to_wrap)
-        if hasattr(original, "handle_out_argument") and not hasattr(
-            to_wrap, "handle_out_argument"
-        ):
-            to_wrap = handle_out_argument(to_wrap)
-        if hasattr(original, "handle_nestable") and not hasattr(
-            to_wrap, "handle_nestable"
-        ):
-            to_wrap = handle_nestable(to_wrap)
+        # set attributes
+        for attr in original.__dict__.keys():
+            # private attribute or decorator
+            if attr.startswith("__") or hasattr(ivy, attr):
```

Comment:
we use a single "_" for private attributes and functions, and so we should change to `attr.startswith("_")`